### DISCUSSION TITLE
How to make binary available to all workspaces?

### DISCUSSION THREAD

**Discussion by xaviergmail** [yarnpkg/berry#6217] [Q&A]
I have a yarn workspace which specifes
```json
  "dependencies": {
    "serverless": "^3.38.0",
    ...
  },
  "workspaces": [
    "sample-workspace",
    ...
  ]
```
 
I then have a `deploy` script in `sample-workspace/package.json`:
```json-
    "deploy": "serverless deploy ... && serverless deploy ..."
```

Running `yarn deploy` inside `sample-workspace/` doesn't work.

I can run `yarn exec serverless` from the root of my repo. However, if I `cd` into `sample-workspace` and try to run `yarn exec serverless`, it can't find the serverless executable installed in the root of the repo either. I assumed this would work. 
 
I have already read [How to share scripts between workspaces?](https://yarnpkg.com/getting-started/qa#how-to-share-scripts-between-workspaces) and none of those solutions are appealing to me.

We have several workspaces and we used to define the serverless dependency in each workspace's devDependencies. I just wanted to hoist that to the root workspace to make it easier to maintain. I'm not interested in having a script to maintain each package.json.

Things I've tried that didn't work:
- setting top-level serverless dependency as a runtime dependency instead of devDependency
- adding `"serverless": "*"` as a peerDependency in child workspaces

Things I've already tried that work:
- installing serverless globally `npm i -g serverless` works but my team doesn't like that even though serverless uses the locally installed serverless just fine
- changing the scripts to use `npx serverless` - this is a tad bit slower

At this point, we're basically only using yarn as a task runner (don't get me started 😂) and my only goal is to avoid having to specify serverless versions in each subproject. Is there another way to make the top-level-only dependency work with child workspace scripts and/or `yarn exec` without cluttering scripts with `npx`?

**Comment by merceyz** [comment]
As the documentation you linked to says; you'll have to use `yarn run -T serverless` in your scripts to explicitly use the binary from the root workspace.
